src/pkg/packager/sources/validate.go

```diff
@@ -83,6 +83,9 @@ func ValidatePackageIntegrity(loaded *layout.PackagePaths, aggregateChecksum str
 	checkedMap[loaded.Signature] = true
 
 	err = lineByLine(checksumPath, func(line string) error {
+		if line == "" {
+			return nil
+		}
 		split := strings.Split(line, " ")
 		sha := split[0]
 		rel := split[1]
```

Comment:
```suggestion
		// If the line is empty (i.e. there is no checksum) simply skip it - this can result from a package with no images/components
		if line == "" {
			return nil
		}

		split := strings.Split(line, " ")
		// If the line is not splitable into two pieces the file is likely corrupted
		if len(split) != 2 {
			return fmt.Errorf("invalid checksum line: %s", line)
		}

		sha := split[0]
		rel := split[1]
```

Reply:
This would be a better more-total fix in case someone has a bad package (since the string could be non-empy but not properly splitable by spaces)

Reply:
Thanks, this makes sense and is a good check to add 🎉